src/Microsoft.Framework.Runtime/ApplicationHostContext.cs

```diff
@@ -107,6 +118,45 @@ public class ApplicationHostContext
             _serviceProvider.Add(typeof(ICompilerOptionsProvider), compilerOptionsProvider);
         }
 
+        private bool IsValidLockFile(LockFile lockFile)
+        {
+            // The lock file should contain dependencies for each framework plus dependencies shared by all frameworks
+            if (lockFile.ProjectFileDependencies.Count != Project.GetTargetFrameworks().Count() + 1)
```

Comment:
`Project.GetTargetFrameworks()` gets called twice and we do `ToDictionary` for a pretty small number of